docs/analysis-techniques.md

```diff
@@ -281,70 +371,64 @@ function getAgeCategory(uint256[] memory age) public pure returns (AGE_CATAGORY[
 
-When verifying a state-machine the assertions to test are mostly related to time-sensitive assertions. For example, if it needs to be verified that function `stepOne()` can never be called before function `stepTwo()`, then we can verify with a state-machine. But also when an invariant (something that needs to always be true) needs to be verified a state-machine can be used. But the nice thing about validating on a state-machine is that is can also be done before writing any code. A state-machine can be created on the functional design for example.
+What if the function calls an external function of an interface and the source code is not known? In this case, the tool could simply return a random value, or it could simply give up and say that symbolic execution is not possible in this case. But this of course ignores the possibility that the external call makes a change in the state variables of the original contract which can influence the flow for the rest of the original function. Or something in between.
 
-Note that when verifying a generated state-machine from the source code, it needs to be trusted that the generated model is correct. This is not the case when verifying on compiled source code. Since there the verification is done on the bytecode that will also be deployed on the blockchain.
+## 6 Static analysis
 
-This [paper](https://hal.archives-ouvertes.fr/hal-02103511/document) shows an example of how a smart contract is can be converted to a state-machine. In this paper the tool [NuSMV](https://nusmv.fbk.eu/) is used to model and verify smart-contracts. When using a state-machine to perform verification on, [temporal logic](https://wickstrom.tech/programming/2021/05/03/specifying-state-machines-with-temporal-logic.html) can be used.
+So far SMT, symbolic execution and fuzzing have been discussed as tools that help to find more bugs than with unit tests alone. These are all analysis tools that check the source code against assertions/requirements that the code should adhere to by running the code. Static analysis is different in that it doesn't run the code, but it looks at the code and tries to find patterns that are known to be problematic.
 
-In another [paper](https://www.fit.vutbr.cz/research/groups/verifit/tools/muse/bytecode08.pdf), symbolic execution was used to generate a state-machine on which temporal properties could be verified. This shown again that verification techniques don't exclude each other.
+For example, [Prettier](https://prettier.io/) (the code formatter) is also static analysis tool. It looks at your code and finds violations of rules defined in the configuration of prettier. Also [Grammarly](https://app.grammarly.com/) can be though of as a static analysis tool. It looks at your text and finds violations of rules defined by the English language.
```

Comment:
Is also static analysis tool => Is also a static analysis tool 
Can be though of => Can be thought of